### PR TITLE
Improve test generation from processes

### PR DESCRIPTION
**Pull request by junkerm**
https://trello.com/c/Zi3UESym/445-testgenerierung-aus-prozessen

### CODE REVIEW

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java [hunk 1]
```diff
@@ -107,59 +215,211 @@ protected void generateTestCases() throws SpecmateException {
...
+				consumer.consumeProcessStart((ProcessStart) currentNode, currentConnection, i);
+			} else if (currentNode instanceof ProcessEnd) {
+				consumer.consumeProcessEnd((ProcessEnd) currentNode, i);
+			}
 		}
 	}
 
+	/**
+	 * Creates a testcase from a a list of variables and conditions reflecting a
+	 * certain path thorugh the model
```
Comment (tobi321) [inline bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: typo: through

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java [hunk 2]
```diff
@@ -107,59 +215,211 @@ protected void generateTestCases() throws SpecmateException {
...
 
-	private String makeAction(IModelNode node) {
-		return node.getName();
+	/** Creates an action text from a process tep */
+	private String makeAction(ProcessStep step) {
+		return step.getName();
 	}
 
-	private String makeAction(ProcessConnection connection) {
+	/** Creates an action text from a process decision and a following conneciton */
```
Comment (tobi321) [inline bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: typo: connection

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java [hunk 3]
```diff
@@ -107,59 +215,211 @@ protected void generateTestCases() throws SpecmateException {
...
+	private String makeAction(ProcessDecision decision, ProcessConnection connection) {
 		if (!hasCondition(connection)) {
 			return "";
 		}
-		return "Establish condition: " + connection.getCondition();
+		return "Establish condition: " + decision.getName() + "=" + connection.getCondition();
 	}
 
-	private String makeCheck(ProcessStep step, ProcessConnection connection) {
+	/** Creates an expected outcome text from a step and a following conneciton */
```
Comment (tobi321) [inline bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: typo: connection

File: web/src/assets/i18n/de.json [hunk 4]
```diff
@@ -38,7 +38,7 @@
 	"Error": "Fehler",
 	"ErrorsandWarnings": "Fehler & Warnungen",
 	"Expected Outcome": "Erwartetes Ergebnis",
-	"ExpectedOutcome": "",
+	"ExpectedOutcome": "Erwartetes Ergebnis",
```
Comment (tobi321) [inline web/src/assets/i18n/de.json]: Do we need ExpectedOutcome and Expected Outcome?

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java [hunk 5]
```diff
@@ -107,59 +215,211 @@ protected void generateTestCases() throws SpecmateException {
...
+	}
+
+	/** Creates a precondition text from a connection */
 	private String makePrecondition(ProcessConnection connection) {
 		return "Establish precondition: " + connection.getCondition();
 	}
 
-	private String makeAction(IModelNode node) {
-		return node.getName();
+	/** Creates an action text from a process tep */
```
Comment (tobi321) [inline bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: typo: step

**Review by tobi321 — CHANGES_REQUESTED** [review]
Apart from the typos, everything works!

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java (hunk 1 above)
Comment (junkerm) [reply bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: done

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java (hunk 2 above)
Comment (junkerm) [reply bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: done

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java (hunk 3 above)
Comment (junkerm) [reply bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: done

File: web/src/assets/i18n/de.json (hunk 4 above)
Comment (junkerm) [reply web/src/assets/i18n/de.json]: done

File: bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java (hunk 5 above)
Comment (junkerm) [reply bundles/specmate-testspecification/src/com/specmate/testspecification/internal/generators/ProcessTestCaseGenerator.java]: done

**Pull request merged**